samples/benchmarks/coremark/README.rst

```diff
@@ -0,0 +1,164 @@
+.. _coremark_sample:
+
+CoreMark
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The sample demonstrates how to run the `CoreMark®`_ benchmark to evaluate the performance of a core.
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+Overview
+********
+
+The sample runs the CoreMark benchmark on the target CPU.
+CoreMark evaluates the CPU efficiency by performing different algorithms, such as state machine, CRC calculation, matrix manipulation, and list processing (find and sort).
+
+To run the CoreMark benchmark on the preferred core, press the corresponding button.
+For the button assignment, see the :ref:`coremark_user_interface` section.
+When the benchmark has completed, you can press the same button to restart it.
+If you want to run the sample upon startup, enable the :kconfig:option:`CONFIG_COREMARK_MODE_FLASH_AND_RUN` Kconfig option.
+
+.. _coremark_user_interface:
+
+User interface
+**************
+
+Each target CPU has an assigned button responsible for starting the benchmark and LED that indicates the ``test in progress`` state:
+
+Button 1:
+   Start the benchmark run on the application core.
+
+Button 2:
+   Start the benchmark run on the network or radio core.
+
+Button 3:
+   Start the benchmark run on the PPR core.
+
+LED 1:
+   Indicates ``test in progress`` on the application core.
+
+LED 2:
+   Indicates ``test in progress`` on the network or radio core.
+
+LED 3:
+   Indicates ``test in progress`` on the PPR core.
+
+.. _coremark_configuration:
+
+Configuration
+*************
+
+|config|
+
+CoreMark runs tests multiple times.
+You can define the number of iterations using the :kconfig:option:`CONFIG_COREMARK_ITERATIONS` Kconfig option.
+By default, the iteration quantity is set to the minimum time required for valid results, which is 10 seconds.
+
+Additional configuration
+========================
+
+Check and configure the following configuration options that are used by the sample:
```

Comment:
I think that CoreMark module configuration should belong to CoreMark module doc (sample should only link there).

Reply:
Currently we kinda duplicate part of the information among these documentation pages (and module's docs does not inform about everything)

Reply:
@annwoj @peknis  what do you think?

Reply:
Well, in case of Memfault docs, for example, the configuration options are introduced in the sample documentation rather than in the integration doc.

Reply:
I am actually wondering if it's a good practice. Maybe we could clearly separate sample-specific options from options for an external component's options (otherwise e.g. integrating external component somewhere else may be tricky)?

Reply:
We might want to introduce only the samplle-specific Kconfig options in the sample doc and the rest in the integration doc.

Reply:
@zycz can we change it here (should not be too difficult). Or maybe we could do it as follow-up action?

Reply:
Yes we can do it. However IMO 90% of user will only use sample. Placing there information about Kconfig options from module has purpose. 

But If @peknis decides that we should separate them then I will follow this decision.

Reply:
I think that if we already decided to separate external module from the sample, then we should maintain proper split of Kconfig options documentation (keep in mind that sample introduces sample-specific options which are unavailable for the external module). We can merge PR as is, but maybe let's add a follow-up ticket to ensure we remember about it.

@peknis, please let me know what do you think about it

Reply:
Also, the external component doc may need to be slightly improved (currently we do not inform about adding code which is a necessary step to integrate CoreMark external module)